clang/lib/CIR/CodeGen/CIRGenExpr.cpp

```diff
@@ -165,6 +165,25 @@ LValue CIRGenFunction::emitDeclRefLValue(const DeclRefExpr *e) {
   return LValue();
 }
 
+mlir::Value CIRGenFunction::evaluateExprAsBool(const Expr *e) {
```

Comment:
This should only be necessary in C, right?  Where 'bool' isn't available.  SO how would this work?

Reply:
Even for C, we want to generate a cir.bool value. The classic codegen does the equivalent, generating IR that produces an `i1` value. For example: https://godbolt.org/z/3KjonvqK1

Reply:
I'm not sure `Bool` is ever used to represent an `i8`. Typically there's a call to `convertForMemory` or something similar that checks for `Bool` and returns an `i8`.

Reply:
Sorry, I meant in 'clang' we consider Bool to be 8 bit(when stored) or 1 bit( in a condition), then have it mean something slightly differently in CIR (which is ONLY the 1 bit representation).  I don't know of anything better though, so I resolved.